RecoTracker/DeDx/plugins/DeDxHitCalibrator.cc

```diff
@@ -0,0 +1,426 @@
+      for (size_t j = 0; j < pixelCluster->pixelADC().size(); j++) {
+        const auto& elec = pixelCluster->pixelADC()[j];
+        delta += elec * MeVPerElectron_;
+        // ( pos , (x , y) )
+        const auto& row = pixelCluster->minPixelRow() + pixelCluster->pixelOffset()[2 * j];
+        const auto& col = pixelCluster->minPixelCol() + pixelCluster->pixelOffset()[2 * j + 1];
+        // Go back to adc
+        const auto& DBgain = pixelCalib_.getGain(detId, col, row);
+        const auto& DBpedestal = pixelCalib_.getPedestal(detId, col, row);
+        if (DBgain > 0.) {
+          double vcal;
+          const auto& theLayer = (detId.subdetId() == 1) ? tkTopo_->pxbLayer(detId) : 0;
+          if (theLayer == 1)
+            vcal = (elec - VCaltoElectronOffset_L1_) / VCaltoElectronGain_L1_;
+          else
+            vcal = (elec - VCaltoElectronOffset_) / VCaltoElectronGain_;
+          const auto adc = DBpedestal + vcal / DBgain;
+          if (adc > pixelSaturationThr_)
+            isSaturated = true;
+        }
+      }
+      // Compute energy
+      const auto& energy = correctEnergy(delta, chipId);
+      if (det != PXF && energy <= 50e-3)
+        continue;
+      // Estimate sigma for cluster
+      const unsigned char& nChannels = pixelCluster->pixelADC().size();
+      const auto esigma = 10e-3 * std::sqrt(nChannels);
+      // Compute charge
+      const auto& charge = pathLength != 0. ? energy / pathLength : 0.;
+      pixelHits.emplace_back(charge, trackMomentum, pathLength, isSaturated, esigma);
+    }
+  }
+}
+
+/*****************************************************************************/
+int DeDxHitCalibrator::getDetId(const DetId& id, const float& thickness) {
+  const auto& subdet = id.subdetId() - 1;
+  if (subdet == TECThin && thickness > 400e-4)
+    return TECThick;
+  return subdet;
+}
+
+/*****************************************************************************/
+float DeDxHitCalibrator::correctEnergy(const float& energy, const ChipId& detId) {
+  const auto& g = dedxCalib_->gain().find(detId);
+  if (applyGain_ && g != dedxCalib_->gain().end())
+    return energy * g->second;
+  return energy;
+}
+
+/*****************************************************************************/
+double DeDxHitCalibrator::getChi2(const std::vector<double>& x,
+                                  const std::vector<std::pair<double, int> >& b,
+                                  const double& coupling,
+                                  const double& iSigma) {
+  const auto& npar = b.size();
+  std::vector<double> y(npar, 0.);
+  for (size_t i = 0; i < npar; i++) {
+    const auto dx = coupling * x[i];
+    if (i >= 1)
+      y[i - 1] += dx;
+    y[i] += x[i] - 2 * dx;
+    if (i + 1 < npar)
+      y[i + 1] += dx;
+  }
+  double chi2(0.);
+  for (size_t i = 0; i < npar; i++) {
+    auto q = (b[i].first - y[i]) * iSigma;
+    if (b[i].second == kIsNormal)
+      chi2 += q * q;
+    if (b[i].second == kIsBelow) {
```

Comment:
```suggestion
    else if (b[i].second == kIsBelow) {
```

Reply:
Addressed in 05c2dc85edef1e8e3a387cd8694941aa4c4e24bd